core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -864,10 +883,15 @@ void performOnReplica(final ShardRouting shard, final String nodeId) {
             }
 
             final DiscoveryNode node = nodes.get(nodeId);
-            transportService.sendRequest(node, transportReplicaAction, replicaRequest, transportOptions, new EmptyTransportResponseHandler(ThreadPool.Names.SAME) {
-                        @Override
-                        public void handleResponse(TransportResponse.Empty vResponse) {
-                            onReplicaSuccess();
+            transportService.sendRequest(node, transportReplicaAction, replicaRequest, transportOptions, new TransportResponseHandler<ShardResponse>() {
```

Comment:
Nit: the indentation is off throughout the body of the anonymous `TransportRequestHandler`.